fl4health/utils/losses.py

```diff
@@ -74,7 +80,20 @@ def compute(self) -> Losses:
         num_losses = len(self.losses_list)
         # Compute average checkpoint and backward losses across list
         checkpoint_loss = torch.sum(torch.FloatTensor([losses.checkpoint for losses in self.losses_list])) / num_losses
-        backward_loss = torch.sum(torch.FloatTensor([losses.backward for losses in self.losses_list])) / num_losses
+        backward_loss: Union[torch.Tensor, Dict[str, torch.Tensor]]
+        if all(isinstance(losses.backward, dict) for losses in self.losses_list):
+            assert isinstance(self.losses_list[0].backward, dict)
+            backward_loss = {
+                key: torch.sum(
+                    torch.FloatTensor(
+                        [losses.backward[key] for losses in self.losses_list if isinstance(losses.backward, dict)]
+                    )
+                )
+                / num_losses
+                for key in self.losses_list[0].backward.keys()
+            }
```

Comment:
IMHO this would be more easily readable as a regular for loop than a dictionary comprehension.